libexec/pyenv-init

```diff
@@ -83,58 +83,51 @@ function help_() {
   ksh )
     profile='~/.profile'
     ;;
-  fish )
-    profile='~/.config/fish/config.fish'
-    ;;
   * )
     profile='your profile'
     ;;
   esac
 
-  { echo "# Add pyenv executable to PATH by adding"
-    echo "# the following to ~/.profile:"
-    echo
+  {
     case "$shell" in
     fish )
+      echo "# Add pyenv executable to PATH by running"
+      echo "# the following interactively:"
+      echo
       echo 'set -Ux PYENV_ROOT $HOME/.pyenv'
       echo 'set -Ux fish_user_paths $PYENV_ROOT/bin $fish_user_paths'
+      echo
+      echo "# Load pyenv automatically by appending"
+      echo "# the following to ~/.config/fish/config.fish:"
+      echo
+      echo 'pyenv init - | source'
+      echo 'status is-login; and pyenv init --path | source'
```

Comment:
`pyenv init --path` needs to come first, and it also only needs to run once as per above.

Reply:
It is already only run for login shells, just like for other shells via .profile, so it's exactly as "once" as for those.

The alternative is to make the code it prints check if anything needs to be done, e.g. via checking if the entries are already in.

E.g. this currently prints

```fish
set -gx PATH '/shims' $PATH
```

If it just did

```fish
contains -- /shims $PATH; or set -gx PATH /shims $PATH
```

it would only add the shims directory if necessary. That way this wouldn't have to be conditioned on login shells. The same could be done for `pyenv init -`.

I can change the order.

Reply:
> It is already only run for login shells...

Okay then.

> I can change the order.

Please do -- because `pyenv init -` does use the result of `pyenv init --path`.